tests/functional/behave_features/common/utils/owner_file_submission.py

```diff
@@ -0,0 +1,351 @@
+        self.secrets.pr_branch = pr_branch
+        self.secrets.index_file = "index.yaml"
+
+        self.set_git_username_email(bot_name, settings.GITHUB_ACTIONS_BOT_EMAIL)
+        self.temp_dir = self.repo_manager.add_worktree()
+
+    def unique_branch(self) -> str:
+        """Returns this instance's unique branch name."""
+        return f"e2e-owners-{self.branch_base_id}"
+
+    def workflow_branches(self, test_name: str) -> (str, str):
+        """Returns the base_branch and pr_branch names for a given test_name."""
+        pretty_test_name = self.test_name.strip().lower().replace(" ", "-")
+        base_branch = (
+            f"{self.unique_branch()}-{pretty_test_name}"
+            if pretty_test_name
+            else f"{self.unique_branch()}-test"
+        )
+        return base_branch, f"{base_branch}-pr-branch"
+
+    def cleanup(self):
+        """Cleans up all artifacts that are created locally and remotely."""
+        logging.debug("RedHatOwnersFileSubmissionE2ETest --> cleanup called!")
+        if self.repo_manager:
+            try:
+                self.repo_manager.cleanup()
+            except Exception as e:
+                logging.error(
+                    f"failed to execute cleanup of the repo_manager with error: {e}"
+                )
+
+        # Teardown step to cleanup branches
+        if self.temp_dir is not None:
+            self.temp_dir.cleanup()
+
+        if self.repo_manager:
+            self.repo_manager.repo.git.worktree("prune")
+
+    def submission_path(self) -> str:  # charts/partner/mycompany/mychartname
+        """Composes the submission path based on the vendor_type, vendor, and chart_name values."""
+        return os.path.join(
+            "charts", self.vendor_category, self.vendor_label, self.chart_name
+        )
+
+    def set_git_username_email(self, username: str, email: str):
+        """Writes the username and email to the repo's .git/config.
+
+        Note that calling this will overwrite repository-local values, which
+        will supercede global values.
+
+        Args:
+            repo: git.Repo instance of the local directory
+            username: git username to set
+            email: git email to set
+        """
+        self.repo_manager.repo.config_writer().set_value(
+            "user", "name", username
+        ).release()
+        self.repo_manager.repo.config_writer().set_value(
+            "user", "email", email
+        ).release()
+
+    def create_and_commit_owners_file(self):
+        self.repo_manager.checkout_branch(self.secrets.base_branch)
+        self.repo_manager.push_branch(self.secrets.test_repo, self.secrets.base_branch)
+        self.repo_manager.checkout_branch(self.secrets.pr_branch)
+        return self._create_and_commit_owners_file(
+            self.submission_path(),
+            self.secrets.pr_branch,
+            self.chart_name,
+            self.vendor_label,
+            self.vendor_name,
+        )
```

Comment:
Note for reviewers:

This calls an internal equivalent because the anticipated caller (behave's implementation.py) will call this function without parameters, whereas we (maintainers) may want to call the internal behavior directly.